src/HotChocolate/Core/src/Validation/DocumentValidatorRule.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using HotChocolate.Language;
+
+namespace HotChocolate.Validation
+{
+    public class DocumentValidatorRule<TVisitor>
+        : IDocumentValidatorRule
+        where TVisitor : DocumentValidatorVisitor, new()
+    {
+        private readonly TVisitor _visitor = new TVisitor();
```

Comment:
We could make this static I guess

Reply:
Though it might also just make things more complex

Reply:
What if the visitor has a state or is not thread-safe?